### DISCUSSION TITLE
Add create_geometry_index table option to a flex config

### DISCUSSION THREAD

**Discussion by nextstopsun** [osm2pgsql-dev/osm2pgsql#1550] [Ideas]
Currently, there is only a `cluster` option to control clustering on geometry index. 
But there are cases when geometry index is not needed at all.
Adding a `create_geometry_index` flag to table options would speed up imports in those cases.

**Comment by joto** [comment]
This has come up several times already. We are looking for some more general solution here. Adding more and more options to enable or disable different parts of the osmp2gsql processing chain will create a huge mess which nobody can understand after a while. If anybody has an idea how to do this, please speak up. We have to keep in mind that osm2pgsql does many different things depending on what command line options are used and what's in the config file. It usually "does the right thing" and it does the "performant" thing, so it knows about a good sequence of doing the processing etc. If we allow the user to fiddle with these things, we loose the ability to change them to make osm2pgsql better for the "normal" use case. We'd have to document what osm2pgsql is doing when so that users can understand what all the new options mean, but then we are stuck with that sequence and so on.

**Comment by nextstopsun** [comment]
This option can default to "true" (or "auto" - same strategy as for `cluster` option), so users don't need either to think about it, or adapt their current configs to this new option. It won't break anything. While IMHO it would be more "performant" not to build gist indexes on geometry, when they're really unnecessary. In some cases this saves quite a lot of time and space.

**Reply by joto** [reply]
You are missing my point. Soon we'll have 20 options, all `auto` and 20 pages of documentation if and when you can use each option and what it does exactly and for each version of osm2pgsql those options change slightly, because we change the way osm2pgsql works underneath. And 20 questions with each new version why the option worked in the last version in situation X and now it doesn't any more.

**Reply by nextstopsun** [reply]
Currently I can't propose any better solution than adding an option. 
@joto Your concern is fully understandable, options add code complexity and should work well together.

I guess flex config was made specifically for providing more options to the user.
Do you agree, that building indexes is quite a resource consuming part of the process? 
So the effect of such an option can't be underestimated. That is why this came up in issues and discussions a few times.
